File: discover_javascript/manipulate_the_data.js
var https = require('https');
var fs = require('fs')

var options = {
    hostname: 'api.github.com',
    path: '/search/repositories?q=language:javascript&sort=stars&order=desc',
    headers: {
	'User-Agent': 'Holberton_School',
	'Authorization': 'token fffa38b10948aa7eff293682308672bc95672ae3'
    }
}


function streamToString(stream, cb) {
    const chunks = [];
    stream.on('data', (chunk) => {
	chunks.push(chunk);
    });
    stream.on('end', () => {
	cb(chunks.join(''));
    });
}


var req = https.request(options, function (res) {
    streamToString(res, (jsonString) => {
	var parsedString = JSON.parse(jsonString);
	var output = parsedString.items.map(function(input) {
	    return input.full_name;
	}).join("\n");
	console.log(output);

	fs.writeFile("/tmp/13", jsonString, (err) => {
	    if(err)
		return console.log(err);
	});
    });

});
req.end();

File: discover_more_python/manipulate_the_data.py
import requests, json

url = 'https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc'
request_headers = {
  'User-Agent': 'Holberton_School',
  'Authorization': 'token fffa38b10948aa7eff293682308672bc95672ae3'
}

#creating a variable with the response of request
r = requests.get(url = url, headers = request_headers)

#converting json data to dictionary
data = json.loads(r.text)

for v in data['items']:
    print v[u'full_name']

#print data['items'][][u'full_name']
#print data["items"]["full_name"]
